src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderFlowValidator.java

```diff
@@ -122,41 +128,80 @@ public Future<Void> checkFundLocationRestrictions(List<CompositePoLine> poLines,
     // TODO: will be removed in scope of https://folio-org.atlassian.net/browse/MODORDERS-981
     // as we'll obtain funds once before the processing and check on emptiness
     if (CollectionUtils.isEmpty(funds)) {
       logger.info("No funds found for PO Line {}, skipping fund-location restrictions check.", poLine.getId());
       return Future.succeededFuture();
     }
 
-    Set<String> validLocations = poLine.getLocations().stream().map(Location::getLocationId).collect(Collectors.toSet());
+    Set<Fund> fundsWithRestrictedLocations = funds.stream().filter(Fund::getRestrictByLocations).collect(Collectors.toSet());
 
-    // Checking fund location against validLocations in POL to identify restricted locations
-    // if there is one, will be stored to put in error as parameter
-    Set<String> restrictedLocations = funds.stream()
-      .filter(Fund::getRestrictByLocations)
-      .flatMap(fund -> fund.getLocationIds().stream())
-      .filter(locationId -> !validLocations.contains(locationId))
-      .collect(Collectors.toSet());
-
-    if (restrictedLocations.isEmpty()) {
+    if (fundsWithRestrictedLocations.isEmpty()) {
+      logger.info("validateLocationRestrictions:: Funds is not restricted by locations");
       return Future.succeededFuture();
     }
 
-    String poLineId = poLine.getId();
-    logger.error("For POL {} locations {} are restricted to be used by all funds", poLineId, restrictedLocations);
-    List<Parameter> parameters = List.of(
-      new Parameter().withKey("poLineId").withValue(poLineId),
-      new Parameter().withKey("poLineNumber").withValue(poLine.getPoLineNumber()),
-      new Parameter().withKey("restrictedLocations").withValue(restrictedLocations.toString())
-    );
-    return Future.failedFuture(new HttpException(422, ErrorCodes.FUND_LOCATION_RESTRICTION_VIOLATION, parameters));
+    return extractRestrictedLocations(poLine, fundsWithRestrictedLocations, requestContext)
+      .compose(restrictedLocations -> {
+        if (restrictedLocations.isEmpty()) {
+          logger.info("validateLocationRestrictions:: restricted locations is not found for poLineId '{}'", poLine.getId());
+          return Future.succeededFuture();
+        }
+
+        String poLineId = poLine.getId();
+        logger.error("For POL {} locations {} are restricted to be used by all funds", poLineId, restrictedLocations);
+        List<Parameter> parameters = List.of(
+          new Parameter().withKey("poLineId").withValue(poLineId),
+          new Parameter().withKey("poLineNumber").withValue(poLine.getPoLineNumber()),
+          new Parameter().withKey("restrictedLocations").withValue(restrictedLocations.toString())
+        );
+        return Future.failedFuture(new HttpException(422, ErrorCodes.FUND_LOCATION_RESTRICTION_VIOLATION, parameters));
+      });
+  }
+
+  private Future<Set<String>> extractRestrictedLocations(CompositePoLine poLine, Set<Fund> fundsWithRestrictedLocations, RequestContext requestContext) {
```

Comment:
one more minor change, should be renamed to extractRestrictedLocationIds, and caller method also should use ids suffix, because we are dealing with ids, not full locations

Reply:
thanks, changed the name